plist_types.go

```diff
@@ -46,6 +46,47 @@ func (p *cfDictionary) sort() {
 	sort.Sort(p)
 }
 
+func (c *cfDictionary) filterNodes(nodes []Node) (keys []string, values []cfValue, filtered []Node) {
+	m := c.toMap()
+	add := map[string]bool{}
+	for _, node := range nodes {
+		key := node.Value()
+		if _, ok := node.(*Annotation); ok {
+			keys = append(keys, key)
+			values = append(values, nil)
+			add[key] = true
+
+		}
+		if _, ok := node.(*MetaNode); ok {
+			if v, ok := m[key]; ok {
+				keys = append(keys, key)
+				values = append(values, v)
+				add[key] = true
+			}
+		}
```

Comment:
Is it possible for node not to be a `MetaNode` or an `Annotation`? It looks like we always append the key and add it to the `add` map?